src/Maestro/Maestro.Web.Tests/DefaultChannelsController20200220Tests.cs

```diff
@@ -274,6 +275,51 @@ public async Task TryToGetOrUpdateNonExistentChannel()
             thirdExpectedFailResult.Should().BeOfType<NotFoundResult>("Getting a default channel for a non-existent default channel should give a not-found type result");
         }
 
+        [Test]
+        public async Task AddDuplicateDefaultChannels()
+        {
+            using TestData data = await BuildDefaultAsync();
+            string channelName = "TEST-CHANNEL-DUPLICATE-ENTRY-SCENARIO";
+            string classification = "TEST-CLASSIFICATION";
+            string repository = "FAKE-REPOSITORY";
+            string branch = "FAKE-BRANCH";
+
+            Channel channel;
+            {
+                var result = await data.ChannelsController.CreateChannel(channelName, classification);
+                channel = (Channel) ((ObjectResult) result).Value;
+            }
+
+            DefaultChannelCreateData testDefaultChannelData = new DefaultChannelCreateData()
+            {
+                Branch = branch,
+                ChannelId = channel.Id,
+                Enabled = true,
+                Repository = repository
+            };
+
+            DefaultChannel defaultChannel;
+            {
+                var result = await data.DefaultChannelsController.Create(testDefaultChannelData);
+                defaultChannel = (DefaultChannel) ((ObjectResult) result).Value;
+            }
+
+            defaultChannel.Should().NotBeNull();
+
+            DefaultChannel defaultChannelDuplicateAdd;
+            {
+                var result = await data.DefaultChannelsController.Create(testDefaultChannelData);
+                defaultChannelDuplicateAdd = (DefaultChannel) ((ObjectResult) result).Value;
+            }
+
+            // Adding the same thing twice should succeed, as well as provide the correct object in return.
+            defaultChannelDuplicateAdd.Should().NotBeNull();
```

Comment:
You might be able to use `defaultChannelDuplicateAdd.Should().BeEquivalentTo(defaultChannel)`, https://fluentassertions.com/objectgraphs/

Reply:
Already tried this, I think to make this work I have to implement the Equals() for the model class and there's a tiny chance that impacts something elsewhere (like sorting)

Reply:
oh nope that was IsSameOrEqualTo, I'll use it.